cli/internal/prune/prune.go

```diff
@@ -190,10 +190,19 @@ func (p *prune) prune(opts *opts) error {
 		}
 	}
 
-	lockfile, err := ctx.Lockfile.Subgraph(lockfileKeys)
+	lockfile, err := ctx.Lockfile.Subgraph(workspaces, lockfileKeys)
 	if err != nil {
 		return errors.Wrap(err, "Failed creating pruned lockfile")
 	}
+
+	if patches := lockfile.Patches(); patches != nil {
+		for _, patch := range patches {
+			if err := fs.CopyFile(&fs.LstatCachedFile{Path: p.base.RepoRoot.Join(patch)}, fullDir.Join(patch).ToStringDuringMigration()); err != nil {
```

Comment:
As of pnpm 7.11.0 conflicting patches just clobber any existing patch

Reply:
Let me rephrase, we're loading the patches from which of these locations?
- `<repo-root>/`
- `<repo-root>/packages/ui-library/`

Since we can define patches at any place in the workspace, I don't know how that resolves into the root lockfile:
```
{
  "pnpm": {
    "patchedDependencies": {
      "express@4.18.1": "patches/express@4.18.1.patch"
    }
  }
}
```

Does it resolve fully-qualified to the workspace path + patch path?

Reply:
We should be loading from the repo root. `pnpm patch-commit` places patches in the workspace root and the paths are encoded in the lockfile relative to the repo root. pnpm hoists every patch and applies it to the entire workspace

Reply:
I'll also add that `pnpm` doesn't respect patches defined in sub-packages so if for example you had your patch listed in `packages/ui-library/package.json` the patch won't appear in the lockfile.

Reply:
Code suggests its always in the same directory as the lock file?
https://github.com/pnpm/pnpm/blob/main/packages/plugin-commands-patching/src/patchCommit.ts#L35

Reply:
Thank y'all for explaining how it works!￼